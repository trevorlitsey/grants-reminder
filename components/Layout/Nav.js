import React from 'react';
import { object } from 'prop-types';
import Link from 'next/link';

import { signOut } from '../../base';

const Nav = (props) => {

	const { user } = props;

	return (
		<div>
			<ul>
				<Link href="/">
					<a><li>GrantsReminder</li></a>
				</Link>
				{user && <a><li onClick={signOut} data-test="sign-out">Sign Out</li></a>}
			</ul>
			<style jsx>{`

			div {
				border-top: 4px solid HSLA(206, 100%, 50%, .8);
			}

			ul {
				border-top: 1px solid HSLA(0, 0%, 87%, .7);
				border-bottom: 1px solid HSLA(0, 0%, 87%, .7);
				display: flex;
				justify-content: space-between;
				margin: 0 auto;
				padding: 0 10px;
				color: black;
				font-weight: bold;
			}
			
      li {
				line-height: 4rem;
        list-style: none;
        margin: 0;
			}
			
			li:hover {
				cursor: pointer;
			}

      a {
        text-decoration: none;
        color: black;
        font-family: "Arial";
      }

      a:hover {
        opacity: 0.6;
      }
    `}</style>
		</div>
	)
}

Nav.propTypes = {
	user: object,
}

export default Nav;