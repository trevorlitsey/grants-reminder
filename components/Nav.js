import React from 'react';
import { object } from 'prop-types';
import Link from 'next/link';

import { signOut } from '../base';

const Nav = (props) => {

	const { user } = props;

	return (
		<div>
			<ul>
				<Link href="/">
					<a><li>Home</li></a>
				</Link>
				{user && <li onClick={signOut} data-test="sign-out">Sign Out</li>}
			</ul>
			<style jsx>{`
			ul {
				border-top: 8px solid HSLA(206, 100%, 50%, .8);
				max-width: 1000px;
				display: flex;
				justify-content: space-between;
				margin: 0 auto 20px;
				padding: 0 10px;
				background: HSLA(200, 6%, 93%, 1.00);
				color: black;
				font-weight: bold;
				// text-transform: uppercase;
			}
			
      li {
				line-height: 4rem;
        list-style: none;
        margin: 0;
			}
			
			li:hover {
				cursor: pointer;
			}

      a {
        text-decoration: none;
        color: black;
        font-family: "Arial";
      }

      a:hover {
        opacity: 0.6;
      }
    `}</style>
		</div>
	)
}

Nav.propTypes = {
	user: object,
}

export default Nav;