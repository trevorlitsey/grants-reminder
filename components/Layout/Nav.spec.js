import React from 'react';
import { shallow } from 'enzyme';

import Nav from './Nav';

const user = {
	uid: 1234567,
	email: 'email@provider.com',
}

describe('nav', () => {

	it('should render without crashing', () => {
		const wrapperSignedOut = renderShallowNav();
		expect(wrapperSignedOut).toMatchSnapshot();
	})

	it('should say SIGN OUT when user is logged in', () => {
		const wrapperSignedIn = renderShallowNav({ user });
		const signOutButton = wrapperSignedIn.find('[data-test="sign-out"]').exists();
		expect(signOutButton).toBeTruthy();
	})

	it('should NOT say sign out when user is logged in', () => {
		const wrapperSignedIn = renderShallowNav();
		const signOutButton = wrapperSignedIn.find('[data-test="sign-out"]').exists();
		expect(signOutButton).toBeFalsy();
	})

})

function renderShallowNav(props = {}) {
	const propsToUser = {
		user: null,
		...props,
	}
	return shallow(<Nav {...propsToUser} />)
}