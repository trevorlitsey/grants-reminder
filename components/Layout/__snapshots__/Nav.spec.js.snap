// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    user={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ul
          className="jsx-3199819151"
        >
          <Link
            href="/"
          >
            <a
              className="jsx-3199819151"
            >
              <li
                className="jsx-3199819151"
              >
                Home
              </li>
            </a>
          </Link>
        </ul>,
        <JSXStyle
          css="div.jsx-3199819151{border-top:4px solid HSLA(206,100%,50%,.8);}ul.jsx-3199819151{border-top:1px solid HSLA(0,0%,87%,.7);border-bottom:1px solid HSLA(0,0%,87%,.7);max-width:1000px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;margin:0 auto;padding:0 10px;color:black;font-weight:bold;}li.jsx-3199819151{line-height:4rem;list-style:none;margin:0;}li.jsx-3199819151:hover{cursor:pointer;}a.jsx-3199819151{text-decoration:none;color:black;font-family:\\"Arial\\";}a.jsx-3199819151:hover{opacity:0.6;}"
          styleId="3199819151"
        />,
      ],
      "className": "jsx-3199819151",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              href="/"
            >
              <a
                className="jsx-3199819151"
              >
                <li
                  className="jsx-3199819151"
                >
                  Home
                </li>
              </a>
            </Link>,
            null,
          ],
          "className": "jsx-3199819151",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                className="jsx-3199819151"
              >
                <li
                  className="jsx-3199819151"
                >
                  Home
                </li>
              </a>,
              "href": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <li
                  className="jsx-3199819151"
                >
                  Home
                </li>,
                "className": "jsx-3199819151",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Home",
                  "className": "jsx-3199819151",
                },
                "ref": null,
                "rendered": "Home",
                "type": "li",
              },
              "type": "a",
            },
            "type": [Function],
          },
          null,
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": "div.jsx-3199819151{border-top:4px solid HSLA(206,100%,50%,.8);}ul.jsx-3199819151{border-top:1px solid HSLA(0,0%,87%,.7);border-bottom:1px solid HSLA(0,0%,87%,.7);max-width:1000px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;margin:0 auto;padding:0 10px;color:black;font-weight:bold;}li.jsx-3199819151{line-height:4rem;list-style:none;margin:0;}li.jsx-3199819151:hover{cursor:pointer;}a.jsx-3199819151{text-decoration:none;color:black;font-family:\\"Arial\\";}a.jsx-3199819151:hover{opacity:0.6;}",
          "styleId": "3199819151",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ul
            className="jsx-3199819151"
          >
            <Link
              href="/"
            >
              <a
                className="jsx-3199819151"
              >
                <li
                  className="jsx-3199819151"
                >
                  Home
                </li>
              </a>
            </Link>
          </ul>,
          <JSXStyle
            css="div.jsx-3199819151{border-top:4px solid HSLA(206,100%,50%,.8);}ul.jsx-3199819151{border-top:1px solid HSLA(0,0%,87%,.7);border-bottom:1px solid HSLA(0,0%,87%,.7);max-width:1000px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;margin:0 auto;padding:0 10px;color:black;font-weight:bold;}li.jsx-3199819151{line-height:4rem;list-style:none;margin:0;}li.jsx-3199819151:hover{cursor:pointer;}a.jsx-3199819151{text-decoration:none;color:black;font-family:\\"Arial\\";}a.jsx-3199819151:hover{opacity:0.6;}"
            styleId="3199819151"
          />,
        ],
        "className": "jsx-3199819151",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                href="/"
              >
                <a
                  className="jsx-3199819151"
                >
                  <li
                    className="jsx-3199819151"
                  >
                    Home
                  </li>
                </a>
              </Link>,
              null,
            ],
            "className": "jsx-3199819151",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  className="jsx-3199819151"
                >
                  <li
                    className="jsx-3199819151"
                  >
                    Home
                  </li>
                </a>,
                "href": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <li
                    className="jsx-3199819151"
                  >
                    Home
                  </li>,
                  "className": "jsx-3199819151",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Home",
                    "className": "jsx-3199819151",
                  },
                  "ref": null,
                  "rendered": "Home",
                  "type": "li",
                },
                "type": "a",
              },
              "type": [Function],
            },
            null,
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": "div.jsx-3199819151{border-top:4px solid HSLA(206,100%,50%,.8);}ul.jsx-3199819151{border-top:1px solid HSLA(0,0%,87%,.7);border-bottom:1px solid HSLA(0,0%,87%,.7);max-width:1000px;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;margin:0 auto;padding:0 10px;color:black;font-weight:bold;}li.jsx-3199819151{line-height:4rem;list-style:none;margin:0;}li.jsx-3199819151:hover{cursor:pointer;}a.jsx-3199819151{text-decoration:none;color:black;font-family:\\"Arial\\";}a.jsx-3199819151:hover{opacity:0.6;}",
            "styleId": "3199819151",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
